src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/FailReason.java

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2022 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.pce.finder;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public class FailReason {
+
+    FailReasonType type;
+    String additionalInfo;
+    Long weight;
```

Comment:
Is this field ever read?

Reply:
Yep, https://github.com/telstra/open-kilda/blob/a82dfb1a6a1f4ace7f469cb347b2314eadd0f3cd/src-java/kilda-pce/src/main/java/org/openkilda/pce/impl/InMemoryPathComputer.java#L171